lib/src/over_react_test/prop_type_utils.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:html';
+import 'dart:js';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/component_base.dart';
+import 'package:over_react_test/jacket.dart';
+import 'package:react/react_client/react_interop.dart' show PropTypes;
+import 'package:test/test.dart';
+
+/// Tests the prop types of a component by calling the props, adding children if necessary, and checking the console
+/// for expected log output.
+void testPropTypesWithUiProps(
+    {@required UiProps componentProps, dynamic childProps, String customErrorMessage, Element mountNode}) {
+  PropTypes.resetWarningCache();
+
+  List<String> consoleErrors = [];
+  JsFunction originalConsoleError = context['console']['error'];
+  context['console']['error'] = new JsFunction.withThis((self, [message, arg1, arg2, arg3,  arg4, arg5]) {
+    consoleErrors.add(message);
+    originalConsoleError.apply([message], thisArg: self);
+  });
+
+  dynamic componentFactory = componentProps();
+
+  if (childProps != null) {
+    if (childProps is List) {
+      componentProps(childProps);
+    } else {
+      componentFactory = componentProps(childProps());
+    }
+  }
+
+  if (mountNode != null) {
+    mount(componentFactory, attachedToDocument: true, mountNode: mountNode);
+  } else {
+    mount(componentFactory, attachedToDocument: true);
+  }
+
+  expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
```

Comment:
if we make the customErrorMessage/s thing into a map or list we should do a for loop here that iterates over each of the errors collected and trys to pair it with the expected ones and fail if the count is differnet or is there are errors that were expected that it didnt find...

Reply:
Let me know of your thoughts on how implemented this general logic..